### PR TITLE
FilesUtil additions

### PR DESCRIPTION
**Pull request by daniel-ohayon**
Added
- a few `COMPOUND_*` keys in `BioMetaKeys` class
- wrapper method for `getMetainfo` in `FilesUtil` to get *full metainfo* of a list of files
- logic to convert dictionary sent from Java into python-client `Metainfo` object
- getters on `MetainfoScalarValue` subclasses mimicking the Java API
- tests for metainfo reading 
- possibility to not supply keys to `FilesUtil.get_metainfo_values_as_strings`, in which case all non-technical keys are fetched

### CODE REVIEW

File: genestack_client/files_util.py [hunk 1]
```diff
@@ -612,3 +612,13 @@ def find_files(self, file_filter, sort_order=SortOrder.DEFAULT, ascending=False,
             raise GenestackException("Invalid sort order")
         return self.invoke('findFiles', file_filter.get_dict(), sort_order, ascending, offset, limit)
 
+    def get_metainfo(self, accessions):
```
Comment (genestack-solomatin) [inline genestack_client/files_util.py]: I don't like this name, it looks like method of file, but do other thing.

File: genestack_client/genestack_metainfo.py [hunk 2]
```diff
@@ -295,3 +295,56 @@ def add_date_time(self, key, time):
...
+    @classmethod
+    def parse_metainfo_from_dict(cls, source_dict):
+        output = Metainfo()
+        for key in source_dict:
+            for scalar_value in map(cls.parse_scalar_value, source_dict[key]):
+                output.add_value(key, scalar_value)
+        return output
+
+    @staticmethod
+    def parse_scalar_value(source_dict):
```
Comment (genestack-solomatin) [inline genestack_client/genestack_metainfo.py]: Can be private

File: genestack_client/metainfo_scalar_values.py [hunk 3]
```diff
@@ -39,22 +39,37 @@ def _xstr(arg):
...
     _TYPE = 'boolean'
 
+    def get_boolean(self):
+        return bool(self.get('value'))
+
 
 class IntegerValue(MetainfoScalarValue):
     _TYPE = 'integer'
 
+    def get_integer(self):
```
Comment (genestack-solomatin) [inline genestack_client/metainfo_scalar_values.py]: Lets sync `get_int` and `get_integer`.  First looks better for me.

Comment (genestack-solomatin) [inline genestack_client/metainfo_scalar_values.py]: May be use properties like in task library?  And for single one we always use `value`

File: genestack_client/files_util.py (hunk 1 above)
Comment (daniel-ohayon) [inline genestack_client/files_util.py]: How about `get_metainfos` ?

Comment (genestack-solomatin) [inline genestack_client/files_util.py]: `collect_metainfo`?

File: genestack_client/genestack_metainfo.py [hunk 4]
```diff
@@ -295,3 +299,56 @@ def add_date_time(self, key, time):
         :rtype: None
         """
         self.add_value(key, DateTimeValue(time))
+
+    @classmethod
+    def parse_metainfo_from_dict(cls, source_dict):
```
Comment (genestack-solomatin) [inline genestack_client/genestack_metainfo.py]: docstring

**Pull request merged**